Modules/OrgUnit/interfaces/class.ilOrgUnitPositionAccessHandler.php

```diff
@@ -0,0 +1,87 @@
+
+	const PERMISSION_VIEW_LEARNING_PROGRESS = 'viewlp';
+	const PERMISSION_VIEW_TEST_RESULTS = 'viewtstr';
+
+
+	/**
+	 * @param int[] $user_ids List of ILIAS-User-IDs which shall be filtered
+	 *
+	 * @return int[] Filtered List of ILIAS-User-IDs
+	 */
+	public function filterUserIdsForCurrentUsersPositionsAndAnyPermission(array $user_ids);
+
+
+	/**
+	 * @param int[] $user_ids    List of ILIAS-User-IDs which shall be filtered
+	 *
+	 * @param int   $for_user_id ID od the user, for which
+	 *
+	 * @return int[] Filtered List of ILIAS-User-IDs
+	 */
+	public function filterUserIdsForUsersPositionsAndAnyPermission(array $user_ids, $for_user_id);
+
+
+	/**
+	 * @param int[]  $user_ids           List of ILIAS-User-IDs which shall be filtered
+	 *
+	 * @param string $permission         See the list of
+	 *                                   available permissions in interface
+	 *                                   ilOrgUnitPositionAccessHandler
+	 *
+	 * @throws \ilOrgUnitAccessException when a unknown permission is used. See the list of
+	 *                                   available permissions in interface
+	 *                                   ilOrgUnitPositionAccessHandler
+	 *
+	 *
+	 * @return int[] Filtered List of ILIAS-User-IDs
+	 */
+	public function filterUserIdsForCurrentUsersPositionsAndPermission(array $user_ids, $permission);
+
+
+	/**
+	 * @param int[]  $user_ids           List of ILIAS-User-IDs which shall be filtered
+	 * @param int    $for_user_id
+	 * @param string $permission         See the list of
+	 *                                   available permissions in interface
+	 *                                   ilOrgUnitPositionAccessHandler
+	 *
+	 * @throws \ilOrgUnitAccessException when a unknown permission is used. See the list of
+	 *                                   available permissions in interface
+	 *                                   ilOrgUnitPositionAccessHandler
+	 *
+	 * @return int[] Filtered List of ILIAS-User-IDs
+	 */
+	public function filterUserIdsForUsersPositionsAndPermission(array $user_ids, $for_user_id, $permission);
+
+
+	/**
+	 * @param string $permission         See the list of
+	 *                                   available permissions in interface
+	 *                                   ilOrgUnitPositionAccessHandler
+	 * @param int[]  $on_user_ids        List of ILIAS-User-IDs
+	 *
+	 * @return bool
+	 */
+	public function isCurrentUserBasedOnPositionsAllowedTo($permission, array $on_user_ids);
+
+
+	/**
+	 * @param int    $which_user_id      Permission check for this ILIAS-User-ID
+	 *
+	 * @param string $permission         See the list of
+	 *                                   available permissions in interface
+	 *                                   ilOrgUnitPositionAccessHandler
+	 * @param int[]  $on_user_ids        List of ILIAS-User-IDs
+	 *
+	 * @return bool
+	 */
+	public function isUserBasedOnPositionsAllowedTo($which_user_id, $permission, array $on_user_ids);
+}
```

Comment:
I think other methods are required for the following use case:

- User has rbac "read" permission for a specific course 
- User has not rbac permission "manage members" for a specific course
- User has position permission "manage members" for a specific course

=> He should see the member administration of the course and the users should be filltered (show only users of "his" org units).

Filtering could be done in the method:

public function filterUserIdsByPositionOfUser($a_user_id, $a_permission, $a_ref_id, array $a_user_ids);
and
public function filterUserIdsByPositionOfCurrentUser($a_permission, $a_ref_id, array $a_user_ids); // global user

Maybe a rbac permission could be an additional parameter: 
e.g 
public function filterUserIdsByRbacOrPositionOfCurrentUser($a_rbac_perm, $a_pos_perm, $a_ref_id, array $a_user_ids) {
if($this->checkAccess($a_rbac_perm,'',$a_ref_id)) {
return $a_user_ids;
}
// do other checks
}

Additionally a method is needed which checks whether a position permission is granted:

public function checkPositionAccess($a_pos_perm, $a_ref_id) or even
public function checkRbacOrPositionPermissionAccess($a_rbac_perm, $a_pos_perm, $a_ref_id)

These methods could be used to decide whether a tab (feature) is shown to a user or not:

- Course manage members subtab
$GLOBALS['DIC']->access()->checkRbacOrPermissionAccess('manage_member,'manage_member',$a_ref_id);

- Show statistics in tests:
$GLOBALS['DIC']->access()->checkRbacOrPermissionAccess('tst_statistics',ilOrgUnitPositionAccessHandler::VIEW_TEST_RESULTS,$a_ref_id);